pkg/trace/sampler/sampler.go

```diff
@@ -68,8 +56,23 @@ const (
 
 	// PriorityUserKeep is the value set by a user to explicitly keep a trace.
 	PriorityUserKeep SamplingPriority = 2
+
+	// 2^64 - 1
+	maxTraceID      = ^uint64(0)
+	maxTraceIDFloat = float64(maxTraceID)
+	// Good number for Knuth hashing (large, prime, fit in int64 for languages without uint64)
+	samplerHasher = uint64(1111111111111111111)
 )
 
+// SampleByRate tells if a trace (from its ID) with a given rate should be sampled
+// Use Knuth multiplicative hashing to leverage imbalanced traceID generators
```

Comment:
I think I know what is meant by `to leverage imbalanced traceID generators`. Consider this alternative:

> SampleByRate returns whether to keep a trace, based on its ID and a sampling rate.
> This assumes that trace IDs are nearly uniformly distributed.